DemoViperTests/Modules/ListRepos/ReposRouterTests.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  ReposRouterTests.swift
+//  DemoViper Tests
+//
+//  Created by Su Van Ho on 13/8/19.
+//  Copyright © 2019 Nimble Co. Ltd. All rights reserved.
+//
+
+import XCTest
+@testable import DemoViper
+
+final class ReposRouterTests: XCTestCase {
+
+    var router: ReposRouter!
+    
+    var viewController: ReposViewController!
+
+    override func setUp() {
+        super.setUp()
+
+        viewController = ReposViewController()
+        viewController.output = ReposViewOutputMock()
+        let navi = UINavigationController(rootViewController: viewController)
+        UIApplication.shared.keyWindow?.rootViewController = navi
+
+        router = ReposRouter()
+        router.view = viewController
+        viewController.loadViewIfNeeded()
+    }
+
+    override func tearDown() {
+        viewController = nil
+        router = nil
+        super.tearDown()
+    }
+
+    func testShowDetail() {
+        // TODOs: Write Test when finish push to another viewController
+        /*
+        let expectation = XCTestExpectation(description: "Wait for animation of push controller")
+        router.detail(with: 0)
+        DispatchQueue.main.asyncAfter(deadline: .now() + 1) {
+            XCTAssertEqual(self.viewController.navigationController?.viewControllers.count, 2)
```

Comment:
In order to test navigation system in VIPER, we will need to create spy class for UINavigationController in test target. Then you will be able to get `pushedViewController` without waiting for animation

```swift
final class SpyNavigationController: UINavigationController {
    
    var pushedViewController: [UIViewController] = []
    
    override func pushViewController(_ viewController: UIViewController, animated: Bool) {
        super.pushViewController(viewController, animated: animated)
        pushedViewController.append(viewController)
    }
    
}
```

Reply:
👍